### PR TITLE
Initial support for disk-backed matrices

### PR DESCRIPTION
**Pull request by PeteHaitch**
We move from using `base::matrix` objects to `DelayedArray::DelayedMatrix` objects for all (user facing) matrix-like data. This provides options for on-disk backing of data, e.g. via the **HDF5Array** package. 

It requires the latest version of **DelayedArray** to pass R CMD check (currently only available in SVN or BioC github mirror), so will fail Travis/rhub until that's available from BioC

Summary of key changes:

- *BSseq*, *BSseqTstat*, *BSseqStat* classes all use `DelayedArray::DelayedMatrix` objects. As noted in `inst/NEWS.rd` this means `updateObject()` needs to be run on all serialized objects from older versions of bsseq (< `v1.11.1`)
- All functions rewritten to support `DelayedArray::DelayedMatrix` objects (or objects that contain them as slots) as **input**. 
- So far, only a few functions re-written that take advantage of this for **output**. These functions gain the `BACKEND` argument following `DelayedArray::realize(x, BACKEND)`. A notable example is `read.bismark()`
- `data/BS.chr22.rda` has been re-saved following `updateObject()`

In summary, there is now basic, complete support for HDF5Array-backed objects. There is ongoing work (targeting Bioc **v3.6**) to improve performance and to provide advice (e.g. a vignette) on working with large data.

### CODE REVIEW

**Comment by PeteHaitch** [comment]
BTW this closes #21 and probably #20
